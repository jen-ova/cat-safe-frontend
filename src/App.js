import React, { useState } from "react";
import PlantSearch from "./components/PlantSearch";
import PlantSearchResults from "./components/PlantSearchResults";
import catPlants from "./data/catplants";

const App = () => {
  const [searchResults, setSearchResults] = useState([]);
  const [hasSearched, setHasSearched] = useState(false);

  const [search, setNewSearch] = useState("");

  const handleSearchChange = (e) => {
    setNewSearch(e.target.value);
  };

  const filtered = !search
    ? catPlants
    : catPlants.filter((plant) =>
        plant.plantName.toLowerCase().includes(search.toLowerCase())
      );

  return (
    <div>
      <PlantSearch
        setSearchResults={setSearchResults}
        setHasSearched={setHasSearched}
        handleSearchChange={handleSearchChange}
        setNewSearch={setNewSearch}
        search={search}
      />
      <PlantSearchResults
        results={searchResults}
        hasSearched={hasSearched}
        filtered={filtered}
      />
    </div>
  );
};

export default App;
