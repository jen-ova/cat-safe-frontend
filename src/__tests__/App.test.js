import React from "react";
import { render } from "@testing-library/react";
import App from "../App";

describe("App", () => {
  const { asFragment } = render(<App />);

  it("renders correctly", () => {
    expect(asFragment()).toMatchSnapshot();
  });
});
