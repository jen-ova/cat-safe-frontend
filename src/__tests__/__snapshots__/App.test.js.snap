// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App renders correctly 1`] = `
<DocumentFragment>
  <div>
    <div>
      <svg
        aria-hidden="true"
        class="svg-inline--fa fa-seedling fa-w-16 icon"
        data-icon="seedling"
        data-prefix="fas"
        focusable="false"
        role="img"
        viewBox="0 0 512 512"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M64 96H0c0 123.7 100.3 224 224 224v144c0 8.8 7.2 16 16 16h32c8.8 0 16-7.2 16-16V320C288 196.3 187.7 96 64 96zm384-64c-84.2 0-157.4 46.5-195.7 115.2 27.7 30.2 48.2 66.9 59 107.6C424 243.1 512 147.9 512 32h-64z"
          fill="currentColor"
        />
      </svg>
      <svg
        aria-hidden="true"
        class="svg-inline--fa fa-cat fa-w-16 icon"
        data-icon="cat"
        data-prefix="fas"
        focusable="false"
        role="img"
        viewBox="0 0 512 512"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M290.59 192c-20.18 0-106.82 1.98-162.59 85.95V192c0-52.94-43.06-96-96-96-17.67 0-32 14.33-32 32s14.33 32 32 32c17.64 0 32 14.36 32 32v256c0 35.3 28.7 64 64 64h176c8.84 0 16-7.16 16-16v-16c0-17.67-14.33-32-32-32h-32l128-96v144c0 8.84 7.16 16 16 16h32c8.84 0 16-7.16 16-16V289.86c-10.29 2.67-20.89 4.54-32 4.54-61.81 0-113.52-44.05-125.41-102.4zM448 96h-64l-64-64v134.4c0 53.02 42.98 96 96 96s96-42.98 96-96V32l-64 64zm-72 80c-8.84 0-16-7.16-16-16s7.16-16 16-16 16 7.16 16 16-7.16 16-16 16zm80 0c-8.84 0-16-7.16-16-16s7.16-16 16-16 16 7.16 16 16-7.16 16-16 16z"
          fill="currentColor"
        />
      </svg>
      <form
        class="search-form"
      >
        <input
          class="search-input"
          type="text"
        />
        <button
          class="search-btn"
          type="submit"
        >
          Is it cat safe?
        </button>
      </form>
    </div>
    <div>
      <p>
        Search results go here
      </p>
    </div>
  </div>
</DocumentFragment>
`;
